core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1399,6 +1399,22 @@ wasm_runtime_dump_mem_consumption(WASMExecEnv *exec_env)
     else
         os_printf("Total aux stack used: no enough info to profile\n");
 
+    /*
+     * Report the native stack usage estimation.
+     *
+     * Unlike the aux stack above, we report the amount unused
+     * because we don't know the stack "bottom".
+     *
+     * Note that this is just about what the runtime itself observed.
+     * It doesn't cover host func implementations, signal handlers, etc.
+     */
+    if (exec_env->native_stack_max_used != (void *)UINTPTR_MAX)
+        os_printf("Native stack left: %zd\n",
+                  exec_env->native_stack_max_used
+                      - exec_env->native_stack_boundary);
```

Comment:
We can get the stack bottom for some platforms, e.g. posix platforms, refer to `pthread_attr_getstack` and https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/shared/platform/common/posix/posix_thread.c#L380

Maybe we can print the actual used size for these platforms.

Reply:
we can. but i'm not sure if it's worth inventing another `os_` interface only for this purpose.

Reply:
I am not sure yet, maybe we can keep your implementation, and discuss it later.